2.cellprofiling/scripts/colocalization_gpu.py

```diff
@@ -125,21 +151,48 @@
             )
             coloc_df = pd.DataFrame(colocalization_features, index=[0])
             coloc_df.columns = [
-                f"{compartment}_{channel1}.{channel2}_{col}" for col in coloc_df.columns
+                f"Colocalization_{compartment}_{channel1}.{channel2}_{col}"
+                for col in coloc_df.columns
             ]
-            coloc_df["object_id"] = object_id
-            coloc_df["channel1"] = channel1
-            coloc_df["channel2"] = channel2
-            coloc_df["compartment"] = compartment
-            coloc_df["image_set"] = image_set_loader.image_set_name
+            coloc_df.insert(0, "object_id", object_id)
+            coloc_df.insert(1, "image_set", image_set_loader.image_set_name)
+            list_of_dfs.append(coloc_df)
+        coloc_df = pd.concat(list_of_dfs, ignore_index=True)
         output_file = pathlib.Path(
             f"../results/{image_set_loader.image_set_name}/Colocalization_{compartment}_{channel1}.{channel2}_features.parquet"
         )
         output_file.parent.mkdir(parents=True, exist_ok=True)
         coloc_df.to_parquet(output_file)
+        coloc_df.head()
 
 
-# In[8]:
+# In[ ]:
 
 
-print(f"Elapsed time: {time.time() - start_time:.2f} seconds")
+end_mem = psutil.Process(os.getpid()).memory_info().rss / 1024**2
+end_time = time.time()
+print(f"Memory usage: {end_mem - start_mem:.2f} MB")
+print("Texture time:")
+print("--- %s seconds ---" % (end_time - start_time))
+print("--- %s minutes ---" % ((end_time - start_time) / 60))
+print("--- %s hours ---" % ((end_time - start_time) / 3600))
+# make a df of the run stats
+run_stats = pd.DataFrame(
+    {
+        "start_time": [start_time],
+        "end_time": [end_time],
+        "start_mem": [start_mem],
+        "end_mem": [end_mem],
+        "time_taken": [(end_time - start_time)],
+        "mem_usage": [(end_mem - start_mem)],
+        "gpu": [True],
+        "well_fov": [well_fov],
+        "feature_type": ["colocalization"],
+    }
+)
+# save the run stats to a file
+run_stats_file = pathlib.Path(
+    f"../results/run_stats/{well_fov}_colocalization_gpu.parquet"
+)
+run_stats_file.parent.mkdir(parents=True, exist_ok=True)
+run_stats.to_parquet(run_stats_file)
```

Comment:
This looked like a repeated pattern. Consider abstracting to a reusable function to reduce the lines of code.